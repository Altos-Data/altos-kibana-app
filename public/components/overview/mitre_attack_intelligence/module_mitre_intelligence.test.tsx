  
/*
 * Wazuh app - ModuleMitreAttackIntelligence Component - Test
 *
 * Copyright (C) 2015-2021 Wazuh, Inc.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Find more information about this on the LICENSE file.
 *
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { ModuleMitreIntelligence } from './module_mitre_intelligence';

jest.mock('../../../react-services', () => ({ 
  WzRequest: () => ({
    apiReq: (method: string, path: string, params: any) => {
      return {
        data: {
          data: {
            affected_items: []
          }
        }
      }
    }
  })
}));

describe('Module Mitre Att&ck intelligence container', () => {
  test('should render a the component', () => {
    const component = shallow(<ModuleMitreIntelligence />);

    expect(component).toMatchSnapshot();
  });

  // test('should render a Health check screen with error', () => {
  //   const component = mount(<HealthCheckTest />);

  //   component.find('CheckResult').at(1).invoke('handleErrors')('setup',['Test error']); // invoke is wrapped with act to await for setState

  //   const callOutError = component.find('EuiCallOut');
  //   expect(callOutError.text()).toBe('Test error');
  // });
});