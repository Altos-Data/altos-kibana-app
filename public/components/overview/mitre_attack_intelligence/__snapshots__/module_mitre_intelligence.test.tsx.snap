// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Mitre Att&ck intelligence container should render a the component 1`] = `
<EuiFlexGroup
  style={
    Object {
      "margin": "0 8px",
    }
  }
>
  <EuiFlexItem>
    <PanelSplit
      content={
        <ModuleMitreAttackIntelligenceRightPanel
          didSearch={false}
          loading={false}
          resourceFilters={Array []}
          results={null}
          searchTerm={Array []}
          selectedResource={null}
        />
      }
      contentProps={
        Object {
          "style": Object {
            "maxHeight": "calc(100vh - 255px)",
            "overflowX": "hidden",
            "overflowY": "auto",
          },
        }
      }
      side={
        <ModuleMitreAttackIntelligenceLeftPanel
          onSearchTermAllResourcesChange={[Function]}
          onSearchTermAllResourcesSearch={[Function]}
          onSelectResource={[Function]}
          selectedResource={null}
        />
      }
      sideProps={
        Object {
          "style": Object {
            "minWidth": 145,
            "overflowX": "hidden",
            "width": "15%",
          },
        }
      }
    />
  </EuiFlexItem>
</EuiFlexGroup>
`;
